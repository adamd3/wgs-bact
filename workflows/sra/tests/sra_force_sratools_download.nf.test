nextflow_workflow {

    name "Test workflow: sra/main.nf"
    script "../main.nf"
    workflow "SRA"
    tag "workflows"
    tag "workflows_sra"
    tag "multiqc_mappings_config"
    tag "sra_fastq_ftp"
    tag "sra_ids_to_runinfo"
    tag "sra_merge_samplesheet"
    tag "sra_runinfo_to_ftp"
    tag "sra_to_samplesheet"
    tag "sra_force_sratools_download"

    test("Parameters: --force_sratools_download") {

        when {
            workflow {
                """
                input[0] = Channel.from('ERR1160846', 'GSE214215', 'SRR12848126')
                """
            }
            params {
                outdir = "results"
                force_sratools_download = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
