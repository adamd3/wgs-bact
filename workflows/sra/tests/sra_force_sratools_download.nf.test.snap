{
    "Parameters: --force_sratools_download": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "samplesheet.csv:md5,7e6a65057fe000f562132ae9e608e87b"
                ],
                "2": [
                    "id_mappings.csv:md5,3e70c965568c59f8c8fbb8e99e7a8b79"
                ],
                "3": [
                    [
                        "multiqc_config.yml:md5,6d4c3e5137704358474330207f5f2b5c",
                        "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9"
                    ]
                ],
                "4": [
                    "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9",
                    "versions.yml:md5,45e4a369df2d3af023b8622aa3f6d5a8",
                    "versions.yml:md5,9b17045ca8bdc272cb3f9d349a81d206",
                    "versions.yml:md5,b21338b74b9a8fe0c7a114f7686c07cd",
                    "versions.yml:md5,b52279f7d6b891a6523d9321f3f85b47"
                ],
                "fastq": [
                    
                ],
                "mappings": [
                    "id_mappings.csv:md5,3e70c965568c59f8c8fbb8e99e7a8b79"
                ],
                "sample_mappings": [
                    [
                        "multiqc_config.yml:md5,6d4c3e5137704358474330207f5f2b5c",
                        "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9"
                    ]
                ],
                "samplesheet": [
                    "samplesheet.csv:md5,7e6a65057fe000f562132ae9e608e87b"
                ],
                "versions": [
                    "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9",
                    "versions.yml:md5,45e4a369df2d3af023b8622aa3f6d5a8",
                    "versions.yml:md5,9b17045ca8bdc272cb3f9d349a81d206",
                    "versions.yml:md5,b21338b74b9a8fe0c7a114f7686c07cd",
                    "versions.yml:md5,b52279f7d6b891a6523d9321f3f85b47"
                ]
            }
        ],
        "timestamp": "2023-10-18T09:43:16.468494798"
    }
}